File: 程式語言 V2答案檔/Go/303.go
package main
import "fmt"

func compute(test int) int {
	for i := 2; i < test; i++{
		if test % i == 0{
			return 0
		}
	}
	return 1
}

func main(){
	var test int
	fmt.Scanln(&test)
	if compute(test) == 0{
		fmt.Printf("%d is not a prime number", test)
	}else{
		fmt.Printf("%d is a prime number", test)
	}
}

File: 程式語言 V2答案檔/PHP/303.php
<?php 
    function compute($val){
        $prime = new SplFixedArray(10000);
        for($i = 2; $i < 10000; $i++){
            if($prime[$i]){
                continue;
            }
            for($j = $i + $i; $j < 10000; $j += $i){
                $prime[$j] = true;
            }
        }
        return $prime[$val] ? 0 : 1;
    }

    $a = intval(fgets(STDIN));
    $is_prime = compute($a);
    printf("%d %s a prime number\n", $a, ($is_prime ? "is" : "is not"));
?>